src/components/SettingsDialog.vue

```diff
@@ -276,40 +273,55 @@ export default {
                         }
                     ),
                 )
-                this.resetInterval = true
                 this.updateInterval = oldVal
             }
         },
         async visibleInfoBlocks(newVal, oldVal) {
-            // Avoid infinite loop on page load and when reseting value after failed submit
-            if (this.resetVisibleInfoBlocks) {
-                this.resetVisibleInfoBlocks = false
+            if (!this.writeChanges) {
                 return
             }
+
             try {
                 const data = visibleInfoBlocksEncode(newVal)
                 await api.config.visibleInfoBlocks.update(data)
-                await this.$store.dispatch("refreshConfig")
```

Comment:
This would immediately update the config in the store and should be maintained. Otherwise changing, e.g., the effect of the "Show cooking time" checkbox would only be visible once a recipe is reloaded and not when is is already visible.